docs/content/en/docs/user-guide/examples/k8s-app-bluegreen-with-istio.md

```diff
@@ -6,4 +6,121 @@ description: >
+
+``` yaml
+apiVersion: networking.istio.io/v1beta1
+kind: DestinationRule
+metadata:
+  name: mesh-istio-bluegreen
+spec:
+  host: mesh-istio-bluegreen
+  subsets:
+  - name: primary
+    labels:
+      pipecd.dev/variant: primary
+  - name: canary
+    labels:
+      pipecd.dev/variant: canary
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+```
+
+- Ensure having at least one Istio's `VirtualService` manifest and all traffic is routed to the `primary`
+
+``` yaml
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: mesh-istio-bluegreen
+spec:
+  hosts:
+    - mesh-istio-bluegreen.pipecd.dev
+  gateways:
+    - mesh-istio-bluegreen
+  http:
+    - route:
+      - destination:
+          host: mesh-istio-bluegreen
+          subset: primary
+        weight: 100
+```
+
+## Enabling blue-green strategy
+
+- Add the following `.pipe.yaml` file into the application directory in the Git repository.
+
+``` yaml
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  pipeline:
+    stages:
+      - name: K8S_CANARY_ROLLOUT
+        with:
+          replicas: 100%
+      - name: K8S_TRAFFIC_ROUTING
+        with:
+          all: canary
+      - name: WAIT_APPROVAL
+      - name: K8S_PRIMARY_ROLLOUT
+      - name: K8S_TRAFFIC_ROUTING
+        with:
+          all: primary
+      - name: K8S_CANARY_CLEAN
+  trafficRouting:
+    method: istio
+    istio:
+      host: mesh-istio-bluegreen
+```
+
+- Send a PR to update the container image version in the Deployment manifest and merge it to trigger a new deployment. PipeCD will plan the deployment with the specified blue-green strategy.
+
+![](/images/example-bluegreen-kubernetes-istio.png)
+<p style="text-align: center;">
+Deployment Details Page
+</p>
+
+- Now you have an automated blue-green deployment for your application. 🎉
+
+## Understanding what happened
+
+In this example, you configured the deployment configuration file (`.pipe.yaml`) to switch all traffic from an old to a new version of the application using Istio's weighted routing feature.
+
+- Stage 1: `K8S_CANARY_ROLLOUT` ensures that the workloads of canary variant (new version) should be deployed. But at this time, they still handle nothing, all traffic is handled by workloads of primary variant.
+The number of workloads (e.g. pod) for canary variant is configured to be 100% of the replicas number of primary varant.
+
+![](/images/example-bluegreen-kubernetes-istio-stage-1.png)
+
+- Stage 2: `K8S_TRAFFIC_ROUTING` ensures that all traffic should be routed to canary variant. Because the `trafficRouting` is configured to use Istio, PipeCD will find Istio's VirtualService resource of this application to control the traffic percentage.
+(You can add a ANALYSIS stage after this to validate the new version. When any nagative impacts are detected, an auto-rollback stage will be executed to switch all traffic back to the primary variant.)
```

Comment:
For those who're interested in the ANAYSIS stage, it could be helpful.
```suggestion
(You can add an [ANALYSIS](/docs/user-guide/automated-deployment-analysis/) stage after this to validate the new version. When any negative impacts are detected, an auto-rollback stage will be executed to switch all traffic back to the primary variant.)
```